example_1/bringup/launch/rrbot.launch.py

```diff
@@ -14,111 +14,78 @@
-            "gui",
-            default_value="true",
-            description="Start RViz2 automatically with this launch file.",
-        )
-    )
-
-    # Initialize Arguments
-    gui = LaunchConfiguration("gui")
-
-    # Get URDF via xacro
-    robot_description_content = Command(
+    return LaunchDescription(
         [
-            PathJoinSubstitution([FindExecutable(name="xacro")]),
-            " ",
-            PathJoinSubstitution(
-                [
-                    FindPackageShare("ros2_control_demo_example_1"),
-                    "urdf",
-                    "rrbot.urdf.xacro",
-                ]
+            DeclareLaunchArgument(
+                "gui",
+                default_value="true",
+                description="Start RViz2 automatically with this launch file.",
+            ),
+            # Control node
+            Node(
+                package="controller_manager",
+                executable="ros2_control_node",
+                parameters=[
+                    PathSubstitution(FindPackageShare("ros2_control_demo_example_1"))
+                    / "config"
+                    / "rrbot_controllers.yaml"
+                ],
+                output="both",
+            ),
+            # robot_state_publisher with robot_description from xacro
+            Node(
+                package="robot_state_publisher",
+                executable="robot_state_publisher",
+                output="both",
+                parameters=[
+                    {
+                        "robot_description": Command(
+                            [
+                                "xacro",
+                                " ",
+                                PathSubstitution(FindPackageShare("ros2_control_demo_example_1"))
+                                / "urdf"
+                                / "rrbot.urdf.xacro",
+                            ]
+                        )
+                    }
+                ],
+            ),
+            Node(
+                package="rviz2",
+                executable="rviz2",
+                name="rviz2",
+                output="log",
+                arguments=[
+                    "-d",
+                    PathSubstitution(FindPackageShare("ros2_control_demo_description"))
+                    / "rrbot/rviz/rrbot.rviz",
+                ],
+                condition=IfCondition(LaunchConfiguration("gui")),
+            ),
+            Node(
+                package="controller_manager",
+                executable="spawner",
+                arguments=["joint_state_broadcaster"],
+            ),
+            Node(
+                package="controller_manager",
+                executable="spawner",
+                parameters=[
+                    PathSubstitution(FindPackageShare("ros2_control_demo_example_1"))
+                    / "config"
+                    / "rrbot_controllers.yaml"
+                ],
+                arguments=["forward_position_controller"],
```

Comment:
the parameters have to be passed as argument to the spawner, not as ros parameter for the spawner itself
```suggestion
                arguments=["forward_position_controller", "--param-file",
                PathSubstitution(FindPackageShare("ros2_control_demo_example_1"))
                    / "config"
                    / "rrbot_controllers.yaml"],
```

Please apply a formatter here, I'm just reviewing from the github UI